File: BOJ/backtracking/9663.py
import time
start=time.time()
def promising(i):
    k=1
    flag=True
    while(k<i and flag):
        if(col[i]==col[k] or abs(col[i]-col[k])==(i-k)):
            flag=False
        k+=1
    return flag


def n_queens(i):
    global cnt
    n=len(col)-1
    if promising(i):
        for j in range(1,n+1):
            col

    
            
        
cnt=0
N=int(input()) #n=4
col=[0 for i in range(N+1)]

n_queens(0)
print(cnt)
print("time:",time.time()-start)

File: BOJ/backtracking/9663.c
#include <stdio.h>
#include <stdlib.h>

int cnt=0;

int promising(int i,int *arr){
    int k=1;
    int flag=1;
    while(k<i && flag){
        if(arr[i]==arr[k] || abs(arr[i]-arr[k])==i-k)
            flag=0;
        k++;
    }
    return flag;
}

void n_queen(int i,int n,int *arr){
    if(i==n){
        cnt+=1;
    }
    if(promising(i,arr)==1){
        for(int j=1; j<=n; j++){
            arr[i+1]=j;
            n_queen(i+1,n,arr);
        }
    }
}

int main(){
    int n=0;
    scanf("%d", &n);
    int *arr = (int*)malloc(sizeof(int)*n+1);
    n_queen(0,n,arr);
    printf("%d", cnt);
    return 0;
}